cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/SSHRemoteProcessController.java

```diff
@@ -88,6 +89,46 @@ public boolean isRunning() throws Exception {
     }
   }
 
+  @Override
+  public RuntimeJobStatus getStatus() throws Exception {
+    // Try to SSH into the host and see if the CDAP runtime process is running or not
+    try (SSHSession session = new DefaultSSHSession(sshConfig)) {
```

Comment:
should not duplicate the code. just make this `return isRunning() ? RUNNING : UNKNOWN`